src/book/03-guides/01-main-codebase/02-maintaining-subsystems.mdx

```diff
@@ -0,0 +1,121 @@
+   ```
+
+5. Wait for the cameras to load and then watch the Vision tab in NUsight. To determine if the output is correct, consult the [vision page](/system/subsystems/vision) for images of the expected output.
+
+If you would like to see the Visual Mesh output in NUsight, you will need to log the data and run it back in NUsight using DataPlayback, since the data is too large to send over a network. See the [DataLogging and DataPlayback guide](/guides/main/data-recording-playback), considering the following:
+
+- Use the [`visualmesh`](https://github.com/NUbots/NUbots/blob/master/roles/visualmesh.role) role to record the data.
+- Set `message.output.CompressedImage` to `true` and add `message.vision.VisualMesh: true` in both [`DataLogging.yaml](https://github.com/NUbots/NUbots/blob/master/module/support/logging/DataLogging/data/config/DataLogging.yaml) and [`DataPlayback.yaml](https://github.com/NUbots/NUbots/blob/master/module/support/logging/DataPlayback/data/config/DataPlayback.yaml).
+- Use the [`playback`](https://github.com/NUbots/NUbots/blob/master/roles/playback.role) role to playback the data, without changes.
+
+### Tuning Detectors
+
+If the Visual Mesh trained with good results but when testing in the system the network did not appear to work very well, you may need to tune the detectors.
+
+[BallDetector.yaml](https://github.com/NUbots/NUbots/blob/master/module/vision/BallDetector/data/config/BallDetector.yaml) and [GoalDetector.yaml](https://github.com/NUbots/NUbots/blob/master/module/vision/GoalDetector/data/config/GoalDetector.yaml) contain the values for tuning the ball and goal detectors respectively.
+
+1. Build and install the `visualmesh` role to a robot.
+2. SSH onto the robot.
+3. Enable NUsight messages on the robot by running
+
+   ```bash
+   nano config/NetworkForwarder.yaml
+   ```
+
+   and set `message.vision.Balls` and `message.vision.Goals` to `true`. This command uses the nano text editor, see this [cheatsheet](https://www.nano-editor.org/dist/latest/cheatsheet.html) for common commands.
+
+4. Run NUsight using `yarn prod` on a computer. Set up NUsight using the [Getting Started page](/guides/main/getting-started) if necessary.
+5. Run `./visualmesh` on the robot.
+6. Alter the configuration file for the detectors while simultenously running the binary on the robot. In a new terminal, SSH onto the robot again and run:
+
+   ```bash
+   nano config/BallDetector.yaml
+   ```
+
+   Change the values and upon saving, the changes will be used immediately by the robot **without** needing to rebuild or rerun the `./visualmesh` binary.
+
+7. Repeat #6 for the goal detector by running
+
+   ```bash
+   nano config/GoalDetector.yaml
+   ```
+
+In general, it might be useful to adjust the `confidence_threshold` on both detectors to improve the results. Other variables may give better results with different values, except for `log_level` and the covariences, `goal_projection_covariance` and `ball_angular_cov`.
```

Comment:
```suggestion
In general, it might be useful to adjust the `confidence_threshold` on both detectors to improve the results. Other variables may give better results with different values, except for `log_level` and the covariances (`goal_projection_covariance` and `ball_angular_cov`).
```